miio/interfaces/vacuum.py

```diff
@@ -0,0 +1,32 @@
+from typing import Dict, Protocol, runtime_checkable
+
+from miio import DeviceStatus
+
+FanspeedPresets = Dict[str, int]
+
+
+@runtime_checkable
+class Vacuum(Protocol):
+    def start(self):
+        """Start vacuuming."""
+
+    def pause(self):
+        """Pause vacuuming."""
+
+    def stop(self):
+        """Stop vacuuming."""
+
+    def home(self):
+        """Return back to base."""
+
+    def find(self):
+        """Request vacuum to play a sound to reveal its location."""
+
+    def status(self) -> DeviceStatus:
+        """Return device status."""
+
+    def set_fan_speed(self, int):
+        """Set fan speed."""
```

Comment:
It would be good to describe argument. Is argument in range 0-100 (percent)? Or can the argument be only one of the "presets"? What the method do if the argument does not match?

Reply:
So this was just based on the current roborock implementation, but neither viomi nor g1vacuum seem to support percentage-based fan speed setting.

The solution would be to have a `set_fan_speed_preset(x)` that accepts "something" that is returned by the implementations' `fan_speed_presets()`. A feature flag & a separate `set_fan_speed_percentage(int)` could be introduced for devices that support arbitrary fan speeds.

I think this is more clearly visible on some fan/air(de)humidifer implementations where the "levels" or "presets" are more common than continuous values between 0 and 100.